File: gradle/gradle_digest_utils.py
#!/usr/bin/python3
import sys
import hashlib

gradle_url = "https://services.gradle.org/distributions/gradle-%s.zip"


def main():
    if len(sys.argv) > 1:
        gradle_version = sys.argv[1].strip()
        complete_url = gradle_url % gradle_version

        md5 = hashlib.md5()
        md5.update(complete_url.encode("utf-8"))

        num = int.from_bytes(md5.digest(), byteorder='big', signed=False)

        # Base36
        alphabet, base36 = ['0123456789abcdefghijklmnopqrstuvwxyz', '']

        while num:
            num, i = divmod(num, 36)
            base36 = alphabet[i] + base36

        print(base36 or alphabet[0])
    else:
        raise ValueError("Needs gradle version!")


if __name__ == "__main__":
    main()

File: gradle/GradleDigestUtils.java
import java.math.BigInteger;
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class GradleDigestUtils {
    private static final String MD5_DIGEST = "MD5";
    private static final Pattern VERSION_PATTERN = Pattern.compile("^(.*?)-(src|bin|all)$");
    private static final String GRADLE_DOWNLOAD_URL = "https://services.gradle.org/distributions/gradle-%s.zip";

    public static void main(String[] args) throws Exception {
        if (args.length > 0) {
            String input = args[0].trim();
            Matcher matcher = VERSION_PATTERN.matcher(input);
            if (matcher.matches()) {
                String url = String.format(GRADLE_DOWNLOAD_URL, input);
                MessageDigest messageDigest = MessageDigest.getInstance(MD5_DIGEST);
                messageDigest.update(url.getBytes(StandardCharsets.UTF_8));
                String digest = new BigInteger(1, messageDigest.digest()).toString(36);
                System.out.println(digest);
            } else {
                System.out.println("Error: Version pattern error!");
                System.exit(-1);
            }
        } else {
            System.out.println("Error: Needs gradle version!");
            System.exit(-1);
        }
    }
}